src/fsharp/range.fsi

```diff
@@ -56,6 +56,8 @@ val mkFileIndexRange : FileIndex -> pos -> pos -> range
 /// This view hides the use of file indexes and just uses filenames 
 val mkRange : string -> pos -> pos -> range
 
+val equals : range -> range -> bool
```

Comment:
Did you consider putting `NoEquality` and `NoComparison` on the range struct?

Also another option is to use `open NonStructuralComparison` more commonly, which at least flushes out places where structural comparison and equality are being used.

Reply:
I've considered doing it, just haven't gotten around to it yet :)

Seems like a fine thing to do

Reply:
So this is actually fairly icky work, but I have a branch that I'll throw up in PR at some point